src/workerd/api/http.c++

```diff
@@ -2111,6 +2112,20 @@ jsg::Promise<jsg::Ref<Response>> fetchImplNoOutputLock(
           IoContext::NULL_CLIENT_CHANNEL, Fetcher::RequiresHostAndProtocol::YES);
     }
 
+    KJ_IF_SOME(dataUrl, DataUrl::tryParse(jsRequest->getUrl())) {
+      // If the URL is a data URL, we need to handle it specially.
+      auto data = dataUrl.getData();
+      auto headers = jsg::alloc<Headers>();
+      headers->set(jsg::ByteString(kj::str("content-type")),
+                   jsg::ByteString(dataUrl.getMimeType().toString()));
+      return js.resolvedPromise(Response::constructor(js, kj::Maybe(kj::heapArray<kj::byte>(data)),
```

Comment:
TODO(now)

* [ ] Given that we're not reusing the `DataUrl` structure here, we can refactor this to move the underlying `data` rather than copying it.